src/WCSimDetectorConstruction.cc

```diff
@@ -359,6 +365,45 @@ G4VPhysicalVolume* WCSimDetectorConstruction::Construct()
   //   logicWCBox->SetVisAttributes(G4VisAttributes::Invisible);
   logicExpHall->SetVisAttributes(G4VisAttributes::Invisible);
 
+  //-----------------------------------------------------
+  //Create BGO Material----------------------------------
+  auto nistManager = G4NistManager::Instance();
+  G4Material* BGO = nistManager->FindOrBuildMaterial("G4_BGO");
```

Comment:
don't you already define the BGO properties in `WCSimConstructMaterials.cc`?

Reply:
I do. I remember redefining it here because I don't know how to link the BGO defined inside `WCSimConstructMaterials.cc` with the material needed for the logicVolume of the BGO.

I've tried now something like `G4Material::GetMaterial("BGO")` but does not work. Do you know exactly how this works?

Reply:
how about define `BGO` in .hh

Reply:
I tried something else that I found more easy to understand, but it is still failing. Could you please take a look?

Reply:
Solved.